app/controllers/katello/api/v2/upstream_subscriptions_controller.rb

```diff
@@ -0,0 +1,46 @@
+require 'katello/resources/candlepin'
+
+module Katello
+  class Api::V2::UpstreamSubscriptionsController < Api::V2::ApiController
+    before_action :find_organization
+
+    resource_description do
+      description "Red Hat subscriptions management platform."
+      api_version 'v2'
+    end
+
+    def_param_group :cp_pagination do
+      param :page, :number, :desc => N_("Page number, starting at 1")
+      param :per_page, :number, :desc => N_("Number of results per page to return")
+    end
+
+    api :GET, "/organizations/:organization_id/upstream_subscriptions",
+      N_("List available subscriptions from Red Hat Subscription Management")
+    param :organization_id, :number, :desc => N_("Organization ID"), :required => true
+    param_group :cp_pagination
+    def index
+      cp_resource.organization = @organization
+      response = cp_resource.get(params: presenter.cp_request_params.merge(subscription_params))
+      if response.code == 200
+        results = presenter.from_pools(JSON.parse(response.body))
+        respond(collection: {results: results})
+      else
+        fail HttpErrors::NotFound, _('Candlepin response error: %s') % response
```

Comment:
returning a NotFound here is not likely what we want

Reply:
does this not raise an exception if the response isn't 200?

Reply:
meaning, is this if/else even required?

Reply:
Oh, so just let it raise the exception. :+1:

Reply:
I confirmed that the API user does see the exception bubble up.

i.e. An issue with key

```
{
    "displayMessage": "Neither PUB key nor PRIV key: header too long",
    "errors": [
        "Neither PUB key nor PRIV key: header too long"
    ]
}
```